x-pack/plugins/security_solution/public/common/components/navigation/index.tsx

```diff
@@ -6,104 +6,99 @@
  */
 
 import React, { useEffect } from 'react';
-import { connect } from 'react-redux';
-import { compose } from 'redux';
 import deepEqual from 'fast-deep-equal';
 
 import { useKibana } from '../../lib/kibana';
-import { RouteSpyState } from '../../utils/route/types';
 import { useRouteSpy } from '../../utils/route/use_route_spy';
-import { makeMapStateToProps } from '../url_state/helpers';
+import { useUrlState } from '../url_state/helpers';
 import { setBreadcrumbs } from './breadcrumbs';
 import { TabNavigation } from './tab_navigation';
-import { SiemNavigationProps, SiemNavigationComponentProps } from './types';
-
-export const SiemNavigationComponent: React.FC<
-  SiemNavigationComponentProps & SiemNavigationProps & RouteSpyState
-> = ({
-  detailName,
-  display,
-  navTabs,
-  pageName,
-  pathName,
-  search,
-  tabName,
-  urlState,
-  flowTarget,
-  state,
-}) => {
-  const {
-    chrome,
-    application: { getUrlForApp },
-  } = useKibana().services;
+import { TabNavigationComponentProps, SecuritySolutionTabNavigationProps } from './types';
+import { RouteSpyState } from '../../utils/route/types';
 
-  useEffect(() => {
-    if (pathName || pageName) {
```

Comment:
primary changes in this file were renaming the component and updating it to purely use hooks. See `useUrlState`. Also the primary nav no longer uses this but uses the hook in this file `use_security_solution_navigation`